apache/incubator-uniffle
# [#1888] feat(server): Reject requireBuffer/sendShuffleData for an application if one of the partitions exceeds the limit

### What changes were proposed in this pull request?

Reject the `requireBuffer` and `sendShuffleData` requests for an application if one of the partitions exceeds the limit.
Introduce a config to limit the maximum of partition size, the client will receive an exception with message to show the partition size and the configured max partition size.

### Why are the changes needed?

Fix: #1888

### Does this PR introduce _any_ user-facing change?

Yes. this PR introduced a new config key `rss.server.huge-partition.size.hard.limit` with default value Long.MAX_VALUE to keep consistent with the previous code.

### How was this patch tested?

Manually tested in our env:
- Configure `rss.server.huge-partition.size.hard.limit` to a small size and wait for the expected exception.
